src/sagemaker/model.py

```diff
@@ -1301,6 +1302,11 @@ def deploy(
                 ENDPOINT_CONFIG_ASYNC_KMS_KEY_ID_PATH,
                 sagemaker_session=self.sagemaker_session,
             )
+            async_inference_config.output_path = resolve_value_from_config(
```

Comment:
will this only affect async inference? Can you add tests to confirm the intelligent default value gets passed?

Reply:
+1